### ISSUE TITLE
Use Snackbar component in reading list

### ISSUE THREAD

**Issue by aitchiss** [forem/forem#15228]
We have a dedicated [Snackbar Preact component](https://github.com/forem/forem/blob/62e150f5afefca59b641d24791dae630dee6c6bb/app/javascript/Snackbar/Snackbar.jsx) which has all styles, interactions and accessibility baked in nicely. However, we're not currently using it in the Reading List area of the app (for the snackbar that appears when archiving/unarchiving a post).

This issue is to refactor `readingList.jsx` to remove the current snackbar code:

https://github.com/forem/forem/blob/228a16f90c139f53317b3e4628ea51c84a271dcc/app/javascript/readingList/readingList.jsx#L190-L196

replacing it with the Snackbar Preact component.

Once this is done we should be able to verify all's well by checking:

- Snackbar appears when a post is archived or unarchived
- Text inside the snackbar matches the action executed
- Snackbar automatically disappears after a period of time
- If you hover a mouse over the snackbar it persists on the screen beyond the standard timeout
- If you use a screen reader, an announcement is made when the snackbar appears, reading aloud the snackbar text

**Comment by aitchiss** [comment]
Closed in https://github.com/forem/forem/pull/15235
